cmake/onnxruntime_providers.cmake

```diff
@@ -431,21 +431,28 @@ if (onnxruntime_USE_DML)
   add_dependencies(onnxruntime_providers_dml ${onnxruntime_EXTERNAL_DEPENDENCIES})
   target_include_directories(onnxruntime_providers_dml PRIVATE ${ONNXRUNTIME_ROOT} ${ONNXRUNTIME_ROOT}/../cmake/external/wil/include)
 
+  if(onnxruntime_target_platform STREQUAL "Win32")
+    set(directml_arch_folder x86)
+  else()
+    set(directml_arch_folder x64)
+  endif()
+
   if (NOT onnxruntime_USE_CUSTOM_DIRECTML)
-    if(NOT onnxruntime_target_platform STREQUAL "x86" AND NOT onnxruntime_target_platform STREQUAL "x64")
```

Comment:
The `Win32` check should go into [onnxruntime_common.cmake#L59](https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime_common.cmake#L59)

Reply:
I'm not sure this needs to be moved, because that check in onnxruntime_common is about generically setting the target platform, and this check is about validating that the target platform works with DML, or rather that is isn't arm or arm64. Since this check seems like it is more specific to a single provider (DML) it would make sense to by in onnxruntime_providers.cmake. Thoughts?